pkg/reconciler/subscription/subscription.go

```diff
@@ -576,17 +571,37 @@ func (r *Reconciler) updateChannelAddSubscriptionV1Beta1(ctx context.Context, ch
 		Generation:    sub.Generation,
 		SubscriberURI: sub.Status.PhysicalSubscription.SubscriberURI,
 		ReplyURI:      sub.Status.PhysicalSubscription.ReplyURI,
+		Delivery:      deliverySpec(sub),
 	}
+
+	// Must not have been found. Add it.
+	channel.Spec.Subscribers = append(channel.Spec.Subscribers, toAdd)
+}
+
+func deliverySpec(sub *v1.Subscription) *eventingduckv1beta1.DeliverySpec {
+
+	var delivery *eventingduckv1beta1.DeliverySpec
+
 	// Only set the deadletter sink if it's not nil. Otherwise we'll just end up patching
 	// empty delivery in there.
 	if sub.Status.PhysicalSubscription.DeadLetterSinkURI != nil {
-		toAdd.Delivery = &eventingduckv1beta1.DeliverySpec{
+		delivery = &eventingduckv1beta1.DeliverySpec{
 			DeadLetterSink: &duckv1.Destination{
 				URI: sub.Status.PhysicalSubscription.DeadLetterSinkURI,
 			},
 		}
 	}
+	if sub.Spec.Delivery != nil && (sub.Spec.Delivery.BackoffDelay != nil || sub.Spec.Delivery.Retry != nil || sub.Spec.Delivery.BackoffPolicy != nil) {
+		if delivery == nil {
+			delivery = &eventingduckv1beta1.DeliverySpec{}
+		}
+		delivery.BackoffPolicy = backoffPolicyV1Beta1(sub.Spec.Delivery.BackoffPolicy)
+		delivery.Retry = sub.Spec.Delivery.Retry
+		delivery.BackoffDelay = sub.Spec.Delivery.BackoffDelay
+	}
+	return delivery
+}
 
-	// Must not have been found. Add it.
-	channel.Spec.Subscribers = append(channel.Spec.Subscribers, toAdd)
+func backoffPolicyV1Beta1(policyType *eventingduckv1.BackoffPolicyType) *eventingduckv1beta1.BackoffPolicyType {
```

Comment:
Can you avoid this func and perform the casting directly where you need it?

Reply:
Done: https://github.com/knative/eventing/pull/4042/commits/42ccc3ee23b34ca6e5b3b2d842e606f74e5158a8